athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaMetadataHandler.java

```diff
@@ -0,0 +1,385 @@
+        partitionSchemaBuilder.addField("preparedStmt", new ArrowType.Utf8());
+    }
+
+    /**
+     * Used to get the partitions that must be read from the request table in order to satisfy the requested predicate.
+     * Here generating the SQL from the request and attaching it as a additional column
+     *
+     * @param blockWriter Used to write rows (partitions) into the Apache Arrow response.
+     * @param request Provides details of the catalog, database, and table being queried as well as any filter predicate.
+     * @param queryStatusChecker A QueryStatusChecker that you can use to stop doing work for a query that has already terminated
+     */
+    @Override
+        public void getPartitions(BlockWriter blockWriter, GetTableLayoutRequest request, QueryStatusChecker queryStatusChecker)
+            throws Exception
+    {
+        logger.info("in getPartitions: "+ request);
+
+        StringBuilder sqlStmt = new StringBuilder();
+        Schema schemaName = request.getSchema();
+        TableName tableName = request.getTableName();
+        Constraints contr  = request.getConstraints();
+
+        try {
+            Connection connection = getOrCreateConn(request);
+
+            //Generate the SQL from user provided query
+            String colNames = schemaName.getFields().stream().map(Field::getName).collect(Collectors.joining(","));
+            sqlStmt.append(colNames);
+
+            String preparedStmt = verticaJdbcSplitQueryBuilder.buildSql(tableName.getSchemaName(), tableName.getTableName(), schemaName, contr, request.getQueryId());
+
+            // write the prepared statement to the partition column created in enhancePartitionSchema
+            blockWriter.writeRows((Block block, int rowNum) -> {
+                boolean matched;
+                matched = block.setValue("preparedStmt", rowNum, preparedStmt);
+                return matched ? 1 : 0;
+            });
+        } catch (SQLException e) {
+           logger.info("Exception in getPartitions: "+ e.getMessage());
+        }
+
+
+    }
+
+    /**
+     * Used to split-up the reads required to scan the requested batch of partition(s).
+     *
+     * Here we execute the SQL on Vertica
+     *
+     * @param allocator Tool for creating and managing Apache Arrow Blocks.
+     * @param request Provides details of the catalog, database, table, andpartition(s) being queried as well as
+     * any filter predicate.
+     * @return A GetSplitsResponse which primarily contains:
+     * 1. A Set<Split> which represent read operations Amazon Athena must perform by calling your read function.
+     * 2. (Optional) A continuation token which allows you to paginate the generation of splits for large queries.
+     */
+    @Override
+    public GetSplitsResponse doGetSplits(BlockAllocator allocator, GetSplitsRequest request)
+    {
+        //ToDo: implement use of a continuation token to use in case of larger queries
+
+        logger.info("doGetSplits: " + request);
```

Comment:
the request is already being logged by the SDK so this is redundant.

Reply:
removed.